### PR TITLE
Feat/boto sns unit tests

### PR DESCRIPTION
**Pull request by major0**
### What does this PR do?

- Add unit tests for boto_sns
- Fix boto_sns to get it to pass all the currently defined unit tests
- Set the stage for testing the boto3 conversion of boto_sns

### Previous Behavior
Previously boto_sns had no unit testing and had no way to validate that it could perform basic operations against SNS topics/subscriptions.

### New Behavior
- Add unit tests for boto_sns
- Fix returning more than 100 topics in `get_all_topics()`
- Fix returning more than 100 subscriptions in `get_all_subscriptions_by_topic()`
- Catch exceptions and properly return True/False as expected.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- N/A Docs
- [X] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: changelog/58800.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+Number of bugs in boto_sns to get it to pass newly added unit tests.
```
Comment (waynew) [inline changelog/58800.fixed:1]: **question** could this message be more specific?

Comment (major0) [reply changelog/58800.fixed]: Yuppers.  Fixed it up to list everything, though, towncrier doesn't make formatting the list all that nice.

**Review by waynew — APPROVED** [review]
👍 

Yeah, I'd rather have a list that's got everything than something a bit higher level. Though one might say this could be an indication that future fixes could be broken up a bit more or something 🙃  (but I think this is fine!)

**Review by cmcmarrow — APPROVED** [review]

**Comment by Ch3LL** [comment]
@major0 looks like there are some failing tests here.

**Comment by Ch3LL** [comment]
@major0 are you able to come back to this one and fix the tests?

**Comment by Ch3LL** [comment]
Closing due to inactivity. Please let me know if you want me to re-open this or feel free to re-open a PR against master with fixes to the failing tests.

**Pull request closed without merging**